### ISSUE TITLE
zfs create: permission denied with zfs 2.2.2 (works with 2.1.14)

### ISSUE THREAD

**Issue by jibanes** [openzfs/zfs#15736]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Debian
Distribution Version	| 12
Kernel Version	| 6.1.70
Architecture	| x86_64
OpenZFS Version	| 2.2.2

### Describe the problem you're observing

zfs create: permission denied with zfs 2.2.2 (works with 2.1.14)

### Describe how to reproduce the problem

```
$ uname -a
Linux xxx 6.1.70 #1 SMP PREEMPT_DYNAMIC Wed Jan  3 18:33:02 PST 2024 x86_64 GNU/Linux
$ zpool version
zfs-2.1.14-1
zfs-kmod-2.1.14-1
$ sudo zpool create -f tank wwn-0x5002538f334514e9
$ sudo zfs create tank/test
$ zfs list tank/test
NAME        USED  AVAIL     REFER  MOUNTPOINT
tank/test    24K  7.14T       24K  /tank/test
```
Everything works fine, then I destroy the pool, unload the zfs & spl modules, switch to 2.2.2 and repeat the test:

```
$ zpool version
zfs-2.2.2-1
zfs-kmod-2.2.2-1
$ sudo zpool create -f tank wwn-0x5002538f334514e9
$ sudo zpool status
pool: tank
state: ONLINE
config:
   NAME                      STATE     READ WRITE CKSUM
   tank                      ONLINE       0     0     0
     wwn-0x5002538f334514e9  ONLINE       0     0     0
errors: No known data errors
$ sudo zfs create tank/test
cannot create 'tank/test': permission denied
```
### Include any warning/errors/backtraces from the system logs
none found

**Comment by auralius-dev** [comment]
Issue should be closed, I do not have this issue with ZFS-2.2.3.
If you are able to create datasets @jibanes, please close this issue.

**Comment by jibanes** [comment]
Thanks let me test sometime next week.

**Comment by jibanes** [comment]
Confirmed that this bug is fixed by zfs-2.2.3.
